### PR TITLE
Login flow v2

### PR DESCRIPTION
**Pull request by rullzer**

### CODE REVIEW

File: src/gui/creds/flow2auth.cpp [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    QObject::connect(job, &SimpleNetworkJob::finishedSignal, this, [this](QNetworkReply *reply) {
+        auto jsonData = reply->readAll();
+        QJsonParseError jsonParseError;
+        QJsonObject json = QJsonDocument::fromJson(jsonData, &jsonParseError).object();
+
+        QString pollToken = json.value("poll").toObject().value("token").toString();
+        QString pollEndpoint = json.value("poll").toObject().value("endpoint").toString();
+        QUrl loginUrl = json["login"].toString();
+
+        if (reply->error() != QNetworkReply::NoError || jsonParseError.error != QJsonParseError::NoError
```
Comment (rullzer) [inline src/gui/creds/flow2auth.cpp]: Shouldn't this be moved up ? So we do not even try to parse things if there is an error?

File: src/gui/creds/flow2auth.cpp [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+        ConfigFile cfg;
+        std::chrono::milliseconds polltime = cfg.remotePollInterval();
+        qCInfo(lcFlow2auth) << "setting remote poll timer interval to" << polltime.count() << "msec";
+        _pollTimer.setInterval(polltime.count());
+        QObject::connect(&_pollTimer, &QTimer::timeout, this, &Flow2Auth::slotPollTimerTimeout);
+        _pollTimer.start();
+
+
+		if (!QDesktopServices::openUrl(authorisationLink())) {
```
Comment (rullzer) [inline src/gui/creds/flow2auth.cpp]: indentation is off

File: src/gui/creds/flow2auth.cpp [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+            // TODO: Ask the user to copy and open the link instead of failing here!
```
Comment (rullzer) [inline src/gui/creds/flow2auth.cpp]: Right good point!

File: src/gui/creds/flow2auth.cpp (hunk 1 above)
Comment (misch7) [reply src/gui/creds/flow2auth.cpp]: Then we would have to duplicate more from the code than we're saving ;-)

File: src/gui/creds/flow2auth.cpp (hunk 2 above)
Comment (misch7) [reply src/gui/creds/flow2auth.cpp]: Powered by Visual Studio^^ Fixed now, thanks.

File: src/gui/creds/flow2auth.cpp (hunk 3 above)
Comment (misch7) [reply src/gui/creds/flow2auth.cpp]: UI handles it now (see https://github.com/nextcloud/desktop/pull/1384/commits/24aeb56d0d9be6b65ceb80db9bce667f6e32fdd1) 😼

**Comment by rullzer** [comment]
Lets get this in now. It is pretty solid already and any issues will pop up I assume.

**Comment by misch7** [comment]
Ok, that's fine for me 😸

**Pull request merged**